### PR TITLE
fix: security vulnerabilities using ZAP

### PR DESCRIPTION
**Pull request by sanjay-k1910**
## What?
- Resolved security vulnerabilities of ZAP

### CODE REVIEW

File: src/app/LayoutCommon.astro [hunk 1]
```diff
@@ -17,22 +18,63 @@ envKeys.forEach((item) => {
 const sessionToken = getFromCookies(Astro.cookies, 'session');
 const refreshToken = getFromCookies(Astro.cookies, 'refresh');
 
+const allowedDomains = "'self' https://www.ayanworks.com https://qaapi.credebl.id https://devapi.credebl.id https://fonts.googleapis.com https://avatars.githubusercontent.com"
+
+// Astro.response.headers.set('Content-Security-Policy',"script-src 'self'; frame-ancestors 'self'");
+Astro.response.headers.set('X-Frame-Options', "SAMEORIGIN");
+Astro.response.headers.set('X-CREDEBL',"SSI");
```
Comment (ajile-in) [inline src/app/LayoutCommon.astro:25]: Is this line still needed? I think we can remove it.

File: src/middleware.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
+// export default async (request: Request, context: any) => {
```
Comment (ajile-in) [inline src/middleware.ts:1]: Please remove all the commented and unused code.

**Review by ajile-in — CHANGES_REQUESTED** [review]
Finally, which approach is working? `deployctl.yaml`? or something else?

File: src/middleware.ts (hunk 2 above)
Comment (sanjay-k1910) [reply src/middleware.ts:1]: Yes Sir, I'm using this PR to test the security-related changes on deployed environments.
This PR is not yet in the merge status.

File: src/app/LayoutCommon.astro (hunk 1 above)
Comment (sanjay-k1910) [reply src/app/LayoutCommon.astro]: Removed the unused code.

**Comment by sanjay-k1910** [comment]
> Finally, which approach is working? `deployctl.yaml`? or something else?

Removed the `deployctl.yaml` file and implemented headers in `middleware.ts` file.

**Review by tipusinghaw — APPROVED** [review]
LGTM

**Review by vivekayanworks — APPROVED** [review]

**Pull request merged**